File: razer-sdk/src/test/java/net/booone/razersdk/color/StaticColorTest.java
package net.booone.razersdk.color;

import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.junit.jupiter.api.Assertions.assertThrows;

class StaticColorTest {
    @Test
    void testColorCreation() {
        //given
        final StaticColor color = new StaticColor(0, 123, 255);

        //then
        assertEquals(0, color.red());
        assertEquals(123, color.green());
        assertEquals(255, color.blue());
        assertEquals("StaticColor[red=0, green=123, blue=255]", color.toString());
    }

    @Test
    void testInvalidColors() {
        assertThrows(IllegalArgumentException.class, () -> new StaticColor(-1, 1, 1));
        assertThrows(IllegalArgumentException.class, () -> new StaticColor(256, 1, 1));
        assertThrows(IllegalArgumentException.class, () -> new StaticColor(1, -112345, 1));
        assertThrows(IllegalArgumentException.class, () -> new StaticColor(1, 500, 1));
        assertThrows(IllegalArgumentException.class, () -> new StaticColor(1, 1, -100));
        assertThrows(IllegalArgumentException.class, () -> new StaticColor(1, 1, 4689));
    }
}

File: test/razer-sdk/color/StaticColor.test.ts
import {describe, expect, test} from 'vitest';
import {StaticColor} from '../../../src/razer-sdk/color/StaticColor.js';

describe('StaticColorTest', () => {
    test('testColorCreation', () => {
        //given
        const color = new StaticColor(0, 123, 255);

        //then
        expect(color.red).toBe(0);
        expect(color.green).toBe(123);
        expect(color.blue).toBe(255);
    });

    test('testInvalidColors', () => {
        expect(() => new StaticColor(-1, 1, 1)).toThrow();
        expect(() => new StaticColor(256, 1, 1)).toThrow();
        expect(() => new StaticColor(1, -112345, 1)).toThrow();
        expect(() => new StaticColor(1, 500, 1)).toThrow();
        expect(() => new StaticColor(1, 1, -100)).toThrow();
        expect(() => new StaticColor(1, 1, 4689)).toThrow();
    });
});
